### PR TITLE
Typescript (Complete notification & notification settings modules)

### PR DESCRIPTION
**Pull request by KiranNiranjan**
## Description
Complete notification & notification settings modules

Screenshot 2019-03-22 at 2 25 36 PM

## QA Checklist
- [ ] Unit-Tests
- [ ] Automation-Tests

### CODE REVIEW

File: demo/index.html [hunk 1]
```diff
@@ -281,7 +316,11 @@ <h1>Symphony Electron API Demo</h1>
             document.location.reload();
         });
 
-        window.addEventListener('message', (event) => console.log(event));
+        window.addEventListener('message', (event) => console.log(event.data));
```
Comment (VishwasShashidhar) [inline demo/index.html]: will this print the notification text?

File: src/app/window-handler.ts [hunk 2]
```diff
@@ -75,6 +77,25 @@ export class WindowHandler {
...
+        return {
+            width: 460,
+            height: 360,
+            show: false,
+            modal: true,
+            autoHideMenuBar: true,
+            webPreferences: {
+                sandbox: true,
+                nodeIntegration: false,
+                devTools: false,
```
Comment (VishwasShashidhar) [inline src/app/window-handler.ts]: should we also have `contextIsolation` as true?

File: src/app/window-handler.ts [hunk 3]
```diff
@@ -258,7 +280,7 @@ export class WindowHandler {
                 return this.destroyAllWindow();
             }
 
-            if (this.config.minimizeOnClose) {
+            if (config.getConfigFields([ 'minimizeOnClose' ]).minimizeOnClose) {
```
Comment (VishwasShashidhar) [inline src/app/window-handler.ts]: why do we need this change?

File: src/renderer/notification-handler.ts [hunk 4]
```diff
@@ -62,14 +65,15 @@ export default class NotificationHandler {
      */
     public setupNotificationPosition() {
         // This feature only applies to windows
-        if (isMac) {
+        if (!isMac || !app.isReady()) {
```
Comment (VishwasShashidhar) [inline src/renderer/notification-handler.ts]: on `windows`, this condition will always be true and it'll be returned. Is this correct logic?

File: src/app/window-handler.ts (hunk 3 above)
Comment (KiranNiranjan) [reply src/app/window-handler.ts]: `config.getConfigFields` always returns the latest from `config` file whereas `this.config` is only for initial read

File: src/renderer/notification-handler.ts (hunk 4 above)
Comment (KiranNiranjan) [reply src/renderer/notification-handler.ts]: fixed 😁

File: src/app/window-handler.ts (hunk 2 above)
Comment (KiranNiranjan) [reply src/app/window-handler.ts]: added

**Pull request merged**